hummingbot/strategy/pure_market_making/pure_market_making_v2.pyx

```diff
@@ -295,10 +318,41 @@ cdef class PureMarketMakingStrategyV2(StrategyBase):
                                                                                    market_info,
                                                                                    market_info_to_active_orders.get(market_info, []),
                                                                                    orders_proposal)
+                filtered_proposal = self.c_filter_orders_proposal_for_takers(market_info, filtered_proposal)
                 self.c_execute_orders_proposal(market_info, filtered_proposal)
         finally:
             self._last_timestamp = timestamp
 
+    # To filter out any orders that are going to be taker orders, i.e. buy order price higher than first ask
+    # and sell order price lower than first bid on the order book.
+    cdef object c_filter_orders_proposal_for_takers(self, object market_info, object orders_proposal):
+        cdef :
+            list buy_prices = []
+            list buy_sizes = []
+            list sell_prices = []
+            list sell_sizes = []
+            MarketBase market = market_info.market
+        if orders_proposal.buy_order_sizes[0] > 0:
+            first_ask = market.c_get_price(market_info.trading_pair, True)
+            for i in range(0, len(orders_proposal.buy_order_prices)):
```

Comment:
This is not Pythonic. Please write

```
for price in order_proposal.buy_order_prices:
 ...
```

instead.

Reply:
I understand the Pythonic for loop and that's how I write my loops, in this occasion I need the index (i) so that later I can access the other list (buy_order_sizes) which are coupled to buy_order_prices by indexing.
```
             if first_ask.is_nan() or (orders_proposal.buy_order_prices[i] < first_ask):
                    buy_prices.append(orders_proposal.buy_order_prices[i])
                    buy_sizes.append(orders_proposal.buy_order_sizes[i])
```